packages/reporting-server/README.md

```diff
@@ -65,8 +62,49 @@ postgres://<user>:<password>@<host>/<database>
 
 for more information, see https://tortoise-orm.readthedocs.io/en/latest/databases.html.
 
+## Run the application
+
+Now that you have already elected, configured and created a database and set you configurations. You need to run the migrations to create the database tables. For that we use `Aerich`:
+
+you can install aerich by running the following command:
+
+```bash
+pip3 install reporting-server[aerich]
+```
+
+Once Aerich is installed, we can proceed to apply migrations to create the database tables by running
+
+```bash
+aerich upgrade
+```
+
+After running this command your database should have all the tables defined in the code. Now you can run the application
+
+```bash
+reporting_server
+
+# Or for server reload
+uvicorn --reload rest_server.app:get_app
+```
 
 # Developers
+## Migration
+We are using [aerich](https://github.com/tortoise/aerich) as our database migration tool. That means that the changes to made to the model will not be reflected in the database automatically. You must run the aerich migration command to create a new migration with the changes:
```

Comment:
What happens if we fail to run the migrate command? I'm guessing the db will still be using the old schemas, and we can expect the app to fail. If that is the case, would it be good to automatically run this as part of the app startup? Maybe add a config to enable auto migrate. If it is enabled, the app will perform the migration automatically, if it is disabled, it will try to detect if the schemas are using the correct version, if not, it will exit with a failure immediately.

Reply:
Yeah, we can do that, having the two options seems more dynamic. Most of the time we will want to have the last migrations. The only case that we don't want to do that is in case we want to test our code for supporting older migrations for a time (here it'll be useful to handle the migrations manually with `aerich` on development).  Maybe this is an edge case but let's say you have 5 reporting-severs instances that point to the same database, but you can update just one instance every hour because we have end-users using the app, so you will have 4 outdated and 1 updated, so you cannot update your database until the 5 instances of the reporting-server are up to date, but your new version should support old APIs and structure as well.

Reply:
In the case of updating the database while one or more instance of the server is still running, I think the server's would definitely need some way to determine the current version of the database? I sounds like it could get very complicated if the database could change schema with every transaction, so I'm think the server would need to somehow lock the database from any schema changes and detect the current schema version before it makes any queries.

In any case, multi schema support must be built into the app, so I think it makes more sense to build schema migrations into it as well. Instead of a boolean, the auto migrate option could be a string or int, which specify the schema version that it should migrate to.

Reply:
> In the case of updating the database while one or more instance of the server is still running, I think the server's would definitely need some way to determine the current version of the database? I sounds like it could get very complicated if the database could change schema with every transaction, so I'm think the server would need to somehow lock the database from any schema changes and detect the current schema version before it makes any queries.

But if that the case, and we will block the app until they are on the same version, maybe we should just turn off all the reporting-server instances, upgrade and migrate the app, and turn it on again. Because as you realized it's not an easy path to follow when you're there. 

> In any case, multi schema support must be built into the app, so I think it makes more sense to build schema migrations into it as well. Instead of a boolean, the auto migrate option could be a string or int, which specify the schema version that it should migrate to.

Ok, I'll create a script to do that.

Reply:
> We can do this for development, but it's not the best practice to do it on deployment. Every instance of the application will attempt to migrate the database. The application has permission to perform destructive updates to the database (this can change in the future). Unless we have only one instance deployed.

> Here is an interesting article related to that https://andrewlock.net/deploying-asp-net-core-applications-to-kubernetes-part-7-running-database-migrations/

Related to this, out-of-process migration looks like the better approach for now, so maybe auto migrate doesn't sound like a good idea? Maybe to work better with the recommended deployment strategy, we need an option to perform migration only and immediately exit. We also need a migrate + run app if we want to support starting with a single executable. Then if we want to go further we need a migration detection logic if running multiple instances in single executable mode, or just put in a doc that multiple instances is not supported in single executable mode.